src/UsgsAstroProjectedLsSensorModel.cpp

```diff
@@ -0,0 +1,799 @@
+    std::string aMessage = "Sensor model not supported.";
+    std::string aFunction = "UsgsAstroProjectedLsPlugin::getModelNameFromModelState()";
+    csm::Error csmErr(aErrorType, aMessage, aFunction);
+    throw(csmErr);
+  }
+  return model_name;
+}
+
+//***************************************************************************
+// UsgsAstroLineScannerSensorModel::getModelState
+//***************************************************************************
+std::string UsgsAstroProjectedLsSensorModel::getModelState() const {
+  auto state = stateAsJson(UsgsAstroLsSensorModel::getModelState());
+  state["m_geoTransform"] = m_geoTransform;
+  state["m_projString"] = m_projString;
+  MESSAGE_LOG(
+      spdlog::level::trace,
+      "m_geoTransform: {}, {}, {}, {}, {}, {} "
+      "m_projString: {} ",
+      m_geoTransform[0], 
+      m_geoTransform[1], 
+      m_geoTransform[2], 
+      m_geoTransform[3], 
+      m_geoTransform[4], 
+      m_geoTransform[5], 
+      m_projString);
+  // Use dump(2) to avoid creating the model string as a single long line
+  std::string stateString = getModelName() + "\n" + state.dump(2);
+  return stateString;
+}
+
+//***************************************************************************
+// UsgsAstroLineScannerSensorModel::reset
+//***************************************************************************
+void UsgsAstroProjectedLsSensorModel::reset() {
+  MESSAGE_LOG(spdlog::level::debug, "Running reset()");
+
+  m_geoTransform = std::vector<double>(6, 0.0);
+  m_projString = "";
+}
+
+//*****************************************************************************
+// UsgsAstroProjectedLsSensorModel Constructor
+//*****************************************************************************
+UsgsAstroProjectedLsSensorModel::UsgsAstroProjectedLsSensorModel() {
+  _no_adjustment.assign(UsgsAstroProjectedLsSensorModel::NUM_PARAMETERS, 0.0);
+}
+
+//*****************************************************************************
+// UsgsAstroProjectedLsSensorModel Destructor
+//*****************************************************************************
+UsgsAstroProjectedLsSensorModel::~UsgsAstroProjectedLsSensorModel() {
+  if (m_logger) {
+    m_logger->flush();
+  }
+}
+
+//---------------------------------------------------------------------------
+// Core Photogrammetry
+//---------------------------------------------------------------------------
+
+//***************************************************************************
+// UsgsAstroLsSensorModel::groundToImage
+//***************************************************************************
+csm::ImageCoord UsgsAstroProjectedLsSensorModel::groundToImage(
+    const csm::EcefCoord &ground_pt, double desired_precision,
+    double *achieved_precision, csm::WarningList *warnings) const {
+  
+  PJ_CONTEXT *C = proj_context_create();
+
+  /* Create a projection. */
+  PJ *isdProj = proj_create(C, (m_projString + " +type=crs").c_str());
```

Comment:
what is the  +crs for (sorry if you probably explained this already)

Reply:
To construct a proj string, you just have to tell proj that the proj4 string is a crs. Kinda weird but it errors without it